urllib3/response.py

```diff
@@ -327,9 +384,14 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and
+                       self.length_remaining not in (0, None)):
+                        raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
```

Comment:
This already fires for read() because it's [raised in httplib](https://github.com/python/cpython/blob/2.7/Lib/httplib.py#L597). This patch is covering the edge case of streamed data which httplib never implemented an error for due to wanting to avoid breaking backwards compatibility as noted [here](https://github.com/python/cpython/blob/2.7/Lib/httplib.py#L614).